ethsigner/signing-api/src/main/java/tech/pegasys/ethsigner/core/signing/TransactionSignerFactory.java

```diff
@@ -0,0 +1,23 @@
+/*
+ * Copyright 2019 ConsenSys AG.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+package tech.pegasys.ethsigner.core.signing;
+
+import java.util.Collection;
+import java.util.Optional;
+
+public interface TransactionSignerFactory {
+
+  Optional<TransactionSigner> getSigner(String address);
```

Comment:
Since this is a factory would expect this be createSigner

Reply:
I've renamed it to `TransactionSignerProvider`